### ISSUE TITLE
Python Streaming job failing to drain with BigQueryIO write errors

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21711]
We have a Python Streaming Dataflow job that writes to BigQuery using the `FILE_LOADS` method and `auto_sharding` enabled. When we try to drain the job it fails with the following error,
```

"/usr/local/lib/python3.7/site-packages/apache_beam/io/gcp/bigquery_tools.py", line 1000, in perform_load_job
ValueError: Either a non-empty list of fully-qualified source URIs must be provided via the source_uris
parameter or an open file object must be provided via the source_stream parameter.

```

Our `WriteToBigQuery` configuration,
```

beam.io.WriteToBigQuery(
  table=options.output_table,
  schema=bq_schema,
  create_disposition=beam.io.BigQueryDisposition.CREATE_IF_NEEDED,

 write_disposition=beam.io.BigQueryDisposition.WRITE_APPEND,
  insert_retry_strategy=RetryStrategy.RETRY_ON_TRANSIENT_ERROR,

 method=beam.io.WriteToBigQuery.Method.FILE_LOADS,
  additional_bq_parameters={
    "timePartitioning":
{
      "type": "HOUR",
      "field": "bq_insert_timestamp",
    },
    "schemaUpdateOptions":
["ALLOW_FIELD_ADDITION", "ALLOW_FIELD_RELAXATION"],
  },
  triggering_frequency=120,
  with_auto_sharding=True,
)

```

We are also noticing that the job only fails to drain when there are actual schema updates. If there are no schema updates the job drains without the above error.

Imported from Jira [BEAM-14146](https://issues.apache.org/jira/browse/BEAM-14146). Original Jira may contain additional context.
Reported by: rahuli.

**Comment by kennknowles** [comment]
From the Jira it looks like the job failure was solved but the drain has trouble completing. I don't think we have to block the release on it, even though it does seem important to continue to figure out.

**Comment by kennknowles** [comment]
CC @chamikaramj from the Jira CCs

**Comment by ahmedabu98** [comment]
Fixed by #23710

**Comment by ahmedabu98** [comment]
.close-issue
